Sources/Linq2DynamoDb.DataContext/DataContext.cs

```diff
@@ -220,6 +221,11 @@ public bool IsAssignedToThisClientInstance(IAmazonDynamoDB client)
         }
 
         /// <summary>
+        /// Used to lock while updating the <see cref="_cachedTableDefinitions" /> field
+        /// </summary>
+        private static readonly object CacheTableDefinitionsLock = new object();
+
+        /// <summary>
         /// Table objects are cached here per DynamoDBClient instance.
         /// In order not to reload table metadata between DataContext instance creations.
         /// </summary>
```

Comment:
This is the static instance being referring to, shared by all lightweight instances.